src/internals/terms.md

```diff
@@ -1,11 +1,14 @@
 # Terms
 
+Dependency solving involves handling sets of packages and versions with precision and logic.
+We thus also need precision in our notations.
```

Comment:
The way i look at it, math gives us powerful tools to solve problems that are in a very specific forms built from few primitives. The problem of dependency management itself is very complex, it has too much complexity to tackle algorithmically directly. So instead, we let the user map them down to cs primitives (e.g. virtual packages for features), map them to math primitives (set and boolean operations), use the solution in math space (CDCL/CDNL), map back to CS primitives and let the user transform them into their output format (e.g. Cargo.lock or requirements.txt)

Reply:
Thanks, I did some changes: https://github.com/pubgrub-rs/guide/pull/11/commits/e0002633992dfd7760f12d7345736d791701f533